### DISCUSSION TITLE
Repo cleanup

### DISCUSSION THREAD

**Discussion by nlemoine** [timber/timber#2502] [Ideas]
Hey, 

For readability and management sake, it would be great if the Timber repository would make a fall cleaning.

There's a lot issues/PRs, many of them are probably outdated. What do you think about using one those "stale bot" I see everywhere? I know they can sometimes be a bit "rude" when closing your legitimate issue without warning (we haven't teached them politeness yet) but I guess they are customizable enough to set a fair "stale" policy.

It would motivate us to bring more support to the raised ticket (at least me). Right now, I'm kinda lost in the issues ocean considering the amount of opened issues (this applies for PR too).

[Branches would also need some cleanup](https://github.com/timber/timber/branches/stale) too (IMHO).

Thoughts?

**Comment by Levdbas** [comment]
With hopefully a 2.0 somewhere this year (please)🤞, I think it would be a great idea to do some householding. I went through the oldest issues (#1647 was my latest) and those are the ones I found that probably can be closed because of several reasons.

If this is appreciated and acted upon, I am willing to go through the other issues as well. Really want to do my part in Timber. :)

@nlemoine  @jarednova  @gchtr 

**solved**
- https://github.com/timber/timber/issues/1125
- https://github.com/timber/timber/issues/1366
- https://github.com/timber/timber/issues/1384 solution with @gchtr comment.
- https://github.com/timber/timber/issues/1440
- https://github.com/timber/timber/issues/1475 probably in https://github.com/timber/timber/pull/1492
- https://github.com/timber/timber/issues/1586 
- https://github.com/timber/timber/issues/1645
- https://github.com/timber/timber/issues/1647

**Questions, which might be closed**
- https://github.com/timber/timber/issues/1185
- https://github.com/timber/timber/issues/1297
- https://github.com/timber/timber/issues/1334
- https://github.com/timber/timber/issues/1381 not related
- https://github.com/timber/timber/issues/1396 uses other plugin
- https://github.com/timber/timber/issues/1424 
- https://github.com/timber/timber/issues/1427
- https://github.com/timber/timber/issues/1442
- https://github.com/timber/timber/issues/1471
- https://github.com/timber/timber/issues/1501
- https://github.com/timber/timber/issues/1529
- https://github.com/timber/timber/issues/1530 neeeds a higher PHP version as a fix.
- https://github.com/timber/timber/issues/1560 and probably stale
- https://github.com/timber/timber/issues/1582 uses other plugin which probably needs some custom code.
- https://github.com/timber/timber/issues/1592
- https://github.com/timber/timber/issues/1612 weird use case.

Stale/outdated
- https://github.com/timber/timber/issues/1187
- https://github.com/timber/timber/issues/1282 ?
- https://github.com/timber/timber/issues/1311
- https://github.com/timber/timber/issues/1332
- https://github.com/timber/timber/issues/1336
- https://github.com/timber/timber/issues/1367
- https://github.com/timber/timber/issues/1390 
- https://github.com/timber/timber/issues/1398 cannot reproduce
- https://github.com/timber/timber/issues/1400 cannot reproduce
- https://github.com/timber/timber/issues/1416 outdated.
- https://github.com/timber/timber/issues/1428 I think this was already fixed some time ago. Probably stale.
- https://github.com/timber/timber/issues/1436
https://github.com/timber/timber/issues/1460
https://github.com/timber/timber/issues/1462 
https://github.com/timber/timber/issues/1469
https://github.com/timber/timber/issues/1477
https://github.com/timber/timber/issues/1536 superseded by  https://github.com/timber/timber/issues/2569  ?
https://github.com/timber/timber/issues/1581
https://github.com/timber/timber/issues/1591
https://github.com/timber/timber/issues/1595 never ran into this myself.
https://github.com/timber/timber/issues/1604
https://github.com/timber/timber/issues/1605 never heard back from reporter.
https://github.com/timber/timber/issues/1611

**Comment by gchtr** [comment]
Oh yes, we’re pushing hard for a 2.0 release. And we’re getting there.

> There's a lot issues/PRs, many of them are probably outdated. What do you think about using one those "stale bot" I see everywhere? I know they can sometimes be a bit "rude" when closing your legitimate issue without warning (we haven't teached them politeness yet) but I guess they are customizable enough to set a fair "stale" policy.

Yes, I think we’ve discussed using a stale bot before. We would have to be careful about making it not too aggressive.

From what I can see, there are bots that provide options to set labels that prevent issues from being closed automatically. I think that would help. Because there are quite a few issues that play into a bigger picture in different areas (CDN usage, image handling in general and others). Having these issues around could help us knowing which factors we’d have to consider. But then again, we could also create "overview" issues where we link closed issues.

So if one of you could help setting up a bot, then that would be great 👍.

I think we’re slowly moving away from trying to solve everything in Timber itself and let developers choose third-party libraries to solve certain problems. As Jared stated it nicely [in a recent podcast-appearance](https://www.youtube.com/watch?v=c3GlhhwpKRo&t=1671s), we’re trying to "get out of the way". When we focus more on the core of Timber, we don’t have to solve every issue or feature request. Instead, I guess we should be more strict and have a mindset where we say, if somebody really wanted to have a new feature and feels passionate about it, they could try to implement it themselves. And we could support them along the way. But it shouldn’t be our job to bear the brunt of the work.

So a cleanup is definitely needed. I see that I’m among the few people who has at least a bit of an overview. I will go through that list that you provided, @Levdbas. Thank you sooo much for that initial work 🙌. I appreciate it very much that you want to help here. It’s a very important part of this project. Taking the time to provide a helpful answer helps other find a solution too and will lead to fewer issues and questions in the future.

**Comment by nlemoine** [comment]
Huge thanks @Levdbas for this stale issues summary!

Funny story: I just read this morning [a blog post](https://fvsch.com/stale-bots) about stale bots that argues... against them 😄

While they're pretty good arguments, I'm still convinced they do not apply to every situations and a stale bot might still be useful if set up according our needs. It won't hurt giving it a chance and can be removed anytime if it doesn't fit in the end.
